lang/cs/Org.Apache.REEF.Network/Elastic/Driver/Default/DefaultElasticTaskSetManager.cs

```diff
@@ -0,0 +1,1344 @@
+                }
+
+                IConfiguration baseConf = confBuilder
+                .BindNamedParameter<ElasticServiceConfigurationOptions.DriverId, string>(
+                    GenericType<ElasticServiceConfigurationOptions.DriverId>.Class,
+                    _driverId)
+                .Build();
+
+                IConfiguration mergedTaskConf = Configurations.Merge(taskInfo.TaskConfiguration, baseConf);
+
+                if (taskInfo.IsActiveContextDisposed)
+                {
+                    Log.Log(Level.Warning,
+                        "Task submit for {0} with a non-active context: spawning a new evaluator.", id + 1);
+
+                    if (taskInfo.TaskStatus == TaskState.Failed)
+                    {
+                        _queuedTasks.Enqueue(id + 1);
+                        taskInfo.SetTaskStatus(TaskState.Queued);
+
+                        SpawnNewEvaluator(id);
+                    }
+
+                    return;
+                }
+
+                taskInfo.ActiveContext.SubmitTask(mergedTaskConf);
+                taskInfo.SetTaskStatus(
+                    taskInfo.TaskStatus.IsRecoverable() ? 
+                    TaskState.Recovering : 
+                    TaskState.Submitted);
+            }
+        }
+
+        private void SendToTasks(IList<IElasticDriverMessage> messages, int retry = 0)
+        {
+            foreach (var returnMessage in messages)
+            {
+                if (returnMessage != null)
+                {
+                    var destination = Utils.GetTaskNum(returnMessage.Destination) - 1;
+                    var taskInfo = _taskInfos[destination] ?? throw new ArgumentNullException("Task Info");
+
+                    lock (taskInfo)
+                    {
+                        if (Completed() || Failed())
+                        {
+                            Log.Log(Level.Warning, "Task submit for a completed or failed Task Set: ignoring.");
+                            taskInfo.DisposeTask();
+
+                            return;
+                        }
+                        if (taskInfo.TaskStatus != TaskState.Running ||
+                            taskInfo.TaskRunner == null)
+                        {
+                            var msg = $"Cannot send message to {destination + 1}: Task Status is {taskInfo.TaskStatus}:";
+
+                            if (taskInfo.TaskStatus == TaskState.Submitted && retry < _parameters.Retry)
+                            {
+                                Log.Log(Level.Warning, msg + " Retry");
+                                System.Threading.Tasks.Task.Run(() =>
+                                {
+                                    Thread.Sleep(_parameters.WaitTime);
+                                    SendToTasks(new List<IElasticDriverMessage>() { returnMessage }, retry + 1);
+                                });
+                            }
+                            else if (retry >= _parameters.Retry)
```

Comment:
by the way, `Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling` (already used elsewhere in REEF) has very convenient mechanisms for retrying - look for `RetryPolicy` usage in our code

Reply:
Yes. Added to the major changes list.